include/seqan3/alignment/pairwise/policy/affine_gap_init_policy.hpp

```diff
@@ -0,0 +1,103 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::affine_gap_init_policy.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <tuple>
+
+#include <seqan3/alignment/configuration/align_config_aligned_ends.hpp>
+#include <seqan3/core/algorithm/configuration.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Implements the initialization of the dynamic programming matrix with affine gaps.
+ * \ingroup alignment_policy
+ * \tparam derived_t   The derived alignment algorithm.
+ */
+template <typename derived_t>
+class affine_gap_init_policy
+{
+protected:
+    /*!\name Constructor, destructor and assignment
+     * \brief Defaulted all standard constructor.
+     * \{
+     */
+    constexpr affine_gap_init_policy()                                           noexcept = default;
+    constexpr affine_gap_init_policy(affine_gap_init_policy const &)             noexcept = default;
+    constexpr affine_gap_init_policy(affine_gap_init_policy &&)                  noexcept = default;
+    constexpr affine_gap_init_policy & operator=(affine_gap_init_policy const &) noexcept = default;
+    constexpr affine_gap_init_policy & operator=(affine_gap_init_policy &&)      noexcept = default;
+    ~affine_gap_init_policy()                                                    noexcept = default;
+    //!}
+
+    /*!\brief Initializes the origin of the dynamic programming matrix.
+     * \tparam        cell_t      The underlying cell type.
+     * \tparam        cache_t     The type of the cache.
+     * \param[in,out] active_cell The current cell in the dynamic programming matrix.
+     * \param[in,out] cache       The cache storing hot helper variables.
+     */
+    template <typename cell_t, typename cache_t>
+    constexpr auto init_origin_cell(cell_t & active_cell, cache_t & cache) const noexcept
+    {
+        using std::get;
```

Comment:
what for?

Reply:
ADL means only looking in associated namespaces (and friends and some other things). Writing `using ...` is not ADL, it means "consider the function from this namespace **in addition to ADL**". 
But you aren't call get anywhere anyway. And the structured bindings are explicitly required to not look at the current namespace or any `using ...` you might have done, they are only allowed to consider member get and ADL get.
If it doesn't compile without `using std::get;` than I am pretty sure it is a bug if it does with it.